src/client/components/pages/parts/add-to-collection-modal.js

```diff
@@ -0,0 +1,309 @@
+import * as bootstrap from 'react-bootstrap';
+import CustomInput from '../../../input';
+import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
+import PropTypes from 'prop-types';
+import React from 'react';
+import ReactSelect from 'react-select';
+import SelectWrapper from '../../input/select-wrapper';
+import request from 'superagent';
+
+
+const {Alert, Col, Button, Modal} = bootstrap;
+
+class AddToCollectionModal extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			collectionsAvailable: [],
+			message: {
+				text: null,
+				type: null
+			},
+			selectedCollections: [],
+			show: this.props.show,
+			showCollectionForm: false
+		};
+
+		this.handleClose = this.handleClose.bind(this);
+		this.getCollections = this.getCollections.bind(this);
+		this.handleAddToCollection = this.handleAddToCollection.bind(this);
+		this.toggleRow = this.toggleRow.bind(this);
+		this.handleShowCollectionForm = this.handleShowCollectionForm.bind(this);
+		this.handleShowAllCollections = this.handleShowAllCollections.bind(this);
+		this.handleAddToNewCollection = this.handleAddToNewCollection.bind(this);
+		this.isValid = this.isValid.bind(this);
+	}
+
+	async componentDidUpdate(prevProps) {
+		const collections = await this.getCollections();
+		if (prevProps.show !== this.props.show) {
+			// eslint-disable-next-line react/no-did-update-set-state
+			this.setState({collectionsAvailable: collections, show: this.props.show});
+		}
+	}
+
+	toggleRow(collectionId) {
+		// eslint-disable-next-line react/no-access-state-in-setstate
+		const oldSelected = this.state.selectedCollections;
+		let newSelected;
+		if (oldSelected.find(selectedId => selectedId === collectionId)) {
+			newSelected = oldSelected.filter(selectedId => selectedId !== collectionId);
+		}
+		else {
+			newSelected = oldSelected.push(collectionId) && oldSelected;
+		}
+		this.setState({
+			selectedCollections: newSelected
+		});
+	}
+
+	async getCollections() {
+		// Get all collections of the user (unlikely that a user will have more than 10000 collections
+		const req = await request.get(`/editor/${this.props.userId}/collections/collections?type=${this.props.entityType}&size=10000`);
```

Comment:
Why the `size` here? Is it because we have a small default size?
If so, we might want to think about implementing `size=0` to fetch all the collections?

Your comment made me think about what would happen if there are a lot of collections.
Maybe at the end of the project we can think about using a virtualized scroll.

Reply:
Yes it has default 20 size and also that endpoint uses `fetchPage` and we need to pass a size in it, otherwise it takes up default `10` size.

The modal is scrollable, so if there are alot of collectoins, the modal will become very long...

Reply:
OK, let's cross that bridge when we come to it, then.
There is a virtualized scrolling library already installed that will help us with that (we use it for the long list of languages to select from)